pkg/controllers/vrpq/query_reconciler.go

```diff
@@ -71,18 +82,45 @@ func (q *QueryReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctr
 		return res, err
 	}
 
-	return ctrl.Result{}, q.runListRestorePoints(ctx, req)
+	finder := iter.MakeSubclusterFinder(q.VRec.Client, q.Vdb)
+	pods, err := finder.FindPods(ctx, iter.FindExisting)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	// find a pod to execute the vclusterops API
+	var podIP string
+	for i := range pods.Items {
+		if i != names.GetNMAContainerIndex() {
+			continue
+		}
+		pod := &pods.Items[i]
+		podIP = pod.Status.PodIP
```

Comment:
Finding a pod doesn't mean it is running. There is other state we should be looking at. Is it running, is the NMA container ready. Those are two things we need to check before using this pod as the host to communicate with.

Reply:
Do we need to check to ensure that it's an NMA container? I realized that it has an NMA container if we define the annotation vertica.com/run-nma-in-sidecar: "true". Otherwise, it will be a server container. A server container still has NMA certs to run vclusterops